libraries/AP_Logger/AP_Logger_File.cpp

```diff
@@ -87,12 +87,11 @@ void AP_Logger_File::Init()
         _log_directory = custom_dir;
     }
 
-    hal.scheduler->expect_delay_ms(3000);
+    EXPECT_DELAY(hal, 3000);
```

Comment:
This is a behaviour change as the delay object will remain in scope well past the point we were zeroing the delay here.

Reply:
I'm happy with scope change